ocaml/xapi/xapi_stunnel.ml

```diff
@@ -0,0 +1,198 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Xapi_stdext_threads.Threadext
+module Unixext = Xapi_stdext_unix.Unixext
+
+module D = Debug.Make (struct let name = "stunnel" end)
```

Comment:
I'd make this `xapi_stunnel_server`, to avoid confusion with https://github.com/xapi-project/xen-api-libs-transitional/blob/656e5e750d77a98e6656ac1c37d9309be795b0b6/stunnel/stunnel.ml#L16